src/query-helpers.js

```diff
@@ -1,9 +1,15 @@
 import {prettyDOM} from './pretty-dom'
 import {fuzzyMatches, matches, makeNormalizer} from './matches'
 import {waitForElement} from './wait-for-element'
+import {getConfig} from './config'
 
 function getElementError(message, container) {
-  return new Error([message, prettyDOM(container)].filter(Boolean).join('\n\n'))
+  const customGetElementError = getConfig().customGetElementError
+  const errorMessages = [message, prettyDOM(container)]
+  if (customGetElementError) {
+    return customGetElementError(...errorMessages)
+  }
```

Comment:
I think I'd prefer it if it were this way:

```suggestion
  if (customGetElementError) {
    return customGetElementError(message, container)
  }
  const errorMessages = [message, prettyDOM(container)]
```

That way the developer writing their custom error getter can pretty-ify the container if they want, but they can do something else with it if they'd rather.

Reply:
I was thinking that too, but I wasnt sure.

Reply:
@kentcdodds just updated with that last suggestion and rebased.

Reply:
Thanks for that!

One other thing, with this now I think we can make things even better. It's hard to explain so I hope you don't mind if I make the change myself and you can ask me questions/provide feedback if you have any.

Reply:
Yeah, that refactor makes sense